src/assets/TypeScriptAsset.js

```diff
@@ -0,0 +1,27 @@
+const JSAsset = require('./JSAsset');
+const localRequire = require('../utils/localRequire');
+
+class TypeScriptAsset extends JSAsset {
+  async transform() {
+    this.ast = await this.parse(this.contents);
+    this.isAstDirty = true;
+  }
+
+  async parse(code) {
+    // require typescript, installed locally in the app
+    let typescript = localRequire('typescript', this.name);
+
+    let transpilerOptions = {
+      compilerOptions: {
```

Comment:
Might make sense to use this already to load a project's tsconfig.json file: https://www.npmjs.com/package/tsconfig

If you do, be sure to overwrite `noEmit` to be false, so that people can use `tsc` as a kind of type-linter by setting `noEmit` to true in their tsconfig.json.

Reply:
will def look into this